alf/environments/carla_sensors.py

```diff
@@ -2152,3 +2157,402 @@ def get_current_observation(self, current_frame):
             np.ndarray: 1D vector contains the distance to the frontal obstacle.
         """
         return np.array([self._distance]).astype(np.float32)
+
+
+# ==============================================================================
+# -- Dynamic Object Sensor -----------------------------------------------------
+# ==============================================================================
+
+
+@alf.configurable
+class DynamicObjectSensor(SensorBase):
+    """DynamicObjectSensor.
+        A sensor that perceives the dynamic objects around the ego agent.
+    """
+
+    def __init__(self,
+                 parent_actor,
+                 alf_world,
+                 history_idx=[-16, -11, -6, -1],
+                 max_history_len=20,
+                 object_filter='vehicle.*',
+                 max_object_number=3,
+                 distance_threshold=100,
+                 with_ego_hist=True,
+                 fill_value=None):
+        """
+        Args:
+            parent_actor (carla.Actor): the parent actor of this sensor
+            alf_world (World): the world object keeping all relevant data and
+                some utility functions (e.g., `_get_traffic_light_waypoints`)
+            navigation_sensor (str): the navigation sensor associated with the
+                `parent_actor`
+            history_idx (list[int]): a list of numbers representing the indices
+                of the history information to be rendered for non-ego vehicles.
+                For example, we can set `history_idx=[-1]` for keep only the
+                most recent observation or `history_idx=[-11, -1]` for both the
+                lastest and also the one 10 steps earlier.
+            max_history_len (int): max number of history length preserved
+            object_filter (str): a string representing the type of dynamic
+                objects to be perceived. By default, surrounding vehicles are
+                perceived.
+            max_object_number (int): including ego vehicle
+                (if with_ego_hist is True); otherwise, all are ado vehicles
+            with_ego_hist (bool): whether inlcude ego history
```

Comment:
`hist` can be an ambiguous shorthand, also `max_history_len` did not use the abbreviation. For clarity and consistency, shall we use `with_ego_history` instead?

Reply:
Changed to  ``with_ego_history``.